moveit_core/collision_detection/include/moveit/collision_detection/world_diff.h

```diff
@@ -124,7 +123,7 @@ namespace collision_detection
     World::ObserverHandle observer_handle_;
 
     /* used to unregister the notifier */
-    boost::weak_ptr<World> world_;
+    std::weak_ptr<World> world_;
```

Comment:
We had a short discussion at #206. Nobody seemed very interested (including me), but if you think that is the right approach you can reopen the PR.